packages/flutter/lib/src/widgets/overflow_bar.dart

```diff
@@ -0,0 +1,482 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/material.dart';
+
+/// Defines the horizontal alignment of [OverflowBar] children
+/// when they're laid out in an overflow column.
+///
+/// This value must be interpreted relative to the ambient
+/// [TextDirection].
+enum OverflowBarAlignment {
+  /// Each child is left-aligned for [TextDirection.ltr],
+  /// right-aligned for [TextDirection.rtl].
+  start,
+
+  /// Each child is right-aligned for [TextDirection.ltr],
+  /// left-aligned for [TextDirection.rtl].
+  end,
+
+  /// Each child is horizontally centered.
+  center,
+}
+
+/// A widget that lays out its [children] in a row unless they
+/// "overflow" the available horizontal space, in which case it lays
+/// them out in a column instead.
+///
+/// This widget's width will expand to contain its children and the
+/// specified [spacing] until it overflows. The overflow column will
+/// consume all of the available width.  The [overflowAlignment]
+/// defines how each child will be aligned within the overflow column
+/// and the [overflowSpacing] defines the gap between each child.
+///
+/// The order that the children appear in the horizontal layout
+/// is defined by the [textDirection], just like the [Row] widget.
+/// If the layout overflows, then children's order within their
+/// column is specified by [overflowDirection] instead.
+///
+/// {@tool dartpad --template=stateless_widget_scaffold_center}
+///
+/// This example defines a simple approximation of a dialog
+/// layout, where the layout of the dialog's action buttons are
+/// defined by an [OverflowBar].  The content is wrapped in a
+/// [SingleChildScrollView], so that if overflow occurs, the
+/// action buttons will be still be accessible by scrolling,
+/// no matter how much vertical space is available.
+///
+/// ```dart
+/// Widget build(BuildContext context) {
+///   return Container(
+///     alignment: Alignment.center,
+///     padding: EdgeInsets.all(16),
+///     color: Colors.black.withOpacity(0.15),
+///     child: Material(
+///       color: Colors.white,
+///       elevation: 24,
+///       shape: RoundedRectangleBorder(
+///         borderRadius: BorderRadius.all(Radius.circular(4.0))
```

Comment:
Nit: Demo code should either consistently use or omit the `.0` on doubles

Reply:
👍